text/0041-fixed-point.md

```diff
@@ -0,0 +1,143 @@
+- `fixed.Shape(i_width, f_width, /, *, signed)`: Create a `fixed.Shape` with `i_width` integer bits and `f_width` fractional bits.
+  - The sign bit is not included in `i_width` or `f_width`, so a `fixed.Shape(7, 8, signed=True)` will be 16 bits wide.
+- `fixed.Shape.cast(shape, f_width=0)`: Cast `shape` to a `fixed.Shape` instance.
+- `.i_width`, `.f_width`, `.signed`: Width and signedness properties.
+- `.const(value)`: Create a `fixed.Const` from `value`.
+- `.as_shape()`: Return the underlying `Shape`.
+- `.__call__(target)`: Create a `fixed.Value` over `target`.
+
+`SQ(*args)` is an alias for `fixed.Shape(*args, signed=True)`.
+
+`UQ(*args)` is an alias for `fixed.Shape(*args, signed=False)`.
+
+`fixed.Value` is a `ValueCastable` subclass.
+The following operations are defined on it:
+
+- `fixed.Value(shape, target)`: Create a `fixed.Value` with `shape` over `target`.
+- `fixed.Value.cast(value, f_width=0)`: Cast `value` to a `fixed.Value`.
+- `.i_width`, `.f_width`, `.signed`: Width and signedness properties.
+- `.shape()`: Return the `fixed.Shape` this was created from.
+- `.as_value()`: Return the underlying value.
+- `.eq(value)`: Assign `value`.
+  - If `value` is a `Value`, it'll be assigned directly to the underlying `Value`.
+  - If `value` is an `int` or `float`, it'll be cast to a `fixed.Const` first.
+  - If `value` is a `fixed.Value`, the precision will be extended or rounded as required.
+- `.round(f_width=0)`: Return a new `fixed.Value` with precision changed to `f_width`, rounding as required.
+- `.__add__(other)`, `.__radd__(other)`, `.__sub__(other)`, `.__rsub__(other)`, `.__mul__(other)`, `.__rmul__(other)`: Binary arithmetic operators.
+  - If `other` is a `Value`, it'll be cast to a `fixed.Value` first.
+  - If `other` is an `int`, it'll be cast to a `fixed.Const` first.
+  - If `other` is a `float`: TBD
+  - The result will be a new `fixed.Value` with enough precision to hold any resulting value without rounding or overflowing.
+- `.__lshift__(other)`, `.__rshift__(other)`: Bit shift operators.
+- `.__neg__()`, `.__pos__()`, `.__abs__()`: Unary arithmetic operators.
+- `.__lt__(other)`, `.__le__(other)`, `.__eq__(other)`, `.__ne__(other)`, `.__gt__(other)`, `.__ge__(other)`: Comparison operators.
+
+`fixed.Const` is a `fixed.Value` subclass.
+The following additional operations are defined on it:
+
+- `fixed.Const(value, shape=None)`: Create a `fixed.Const` from `value`. `shape` must be a `fixed.Shape` if specified.
+  - If `value` is an `int` and `shape` is not specified, the smallest shape that will fit `value` will be selected.
+  - If `value` is a `float` and `shape` is not specified, the smallest shape that gives a perfect representation will be selected.
+    If `shape` is specified, `value` will be rounded to the closest representable value first.
+- `.as_integer_ratio()`: Return the value represented as an integer ratio `tuple`.
+- `.as_float()`: Return the value represented as a `float`.
```

Comment:
What happens if it's not representable? At least we need to have `c.as_float(exact=True)` which gives an error in that case.